README.md

```diff
@@ -22,4 +22,32 @@ A Snap plugin is a program that responds to a set of well defined [gRPC](http://
 You will find [example plugins](examples) that cover the basics for writing collector, processor, and publisher plugins in the examples folder.
 
 
-
+## Snap Diagnostics
+Snap plugins using plugin-lib-go can be run independent of Snap to show their current running diagnostics. This diagnostic information includes:
+* Warning if dependencies not met
+* Config policy
+    * Warning if config items required and not provided
+* Collectable metrics
+* Metric catalog
+* Runtime details
+    * Plugin version
+    * RPC type and version
+    * OS, architecture
+    * Golang version
+* How long it took to run each of these diagnostics
+
+### Running Diagnostics
+Running plugin diagnostics is easy! Simply build the plugin, then run the executable `$./build/darwin/x86_64/examples/snap-plugin-collector-rand`. When ran on its own, it will show a warning if a config is required for the plugin to load. 
```

Comment:
replace executable with ./build/${GOOS}/${GOARCH}/<plugin_binary>. An example could be what you have.